{{ cookiecutter.package_name }}/.github/workflows/tox-tests.yml

```
@@ -0,0 +1,104 @@
+# GitHub Actions workflow for testing and continuous integration.
+#
+# This file performs testing using tox and tox.ini to define and configure the test environments.
+
+name: Python Tests
+
+on: [push, pull_request]
+
+jobs:
+
+  # Set up matrix to run tox tests across lists of os, python version, and tox environment
+  matrix_tests:
+    runs-on: ${{ matrix.os }}
+    if: "!contains(github.event.head_commit.message, '[ci skip]')"  # github actions doesn't yet support this natively so add here
+    strategy:
+      matrix:
+        # Github actions supports ubuntu, windows, and macos virtual environments:
+        # https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners
+        #
+        # Only run on ubuntu by default, but can add other os's to the test matrix here.
+        # For example -- os: [ubuntu-latest, macos-latest, windows-latest]
+        os: [ubuntu-latest]
+
+        # Test python 3.7 and 3.8 by default.
+        python-ver: [7, 8]
```

To future proof (in case Python 4 comes out), maybe we also specify major version here.
```suggestion
        python-ver: [3.7, 3.8]
```

---

yeah, i got bit by this already for different reasons so was planning to make this change, anyway.

---

I also prefer the full version number, and the more explicit `python-version` which we are using in the other repos.

---

Also it would be good to add an example of a "manual" matrix, with `include:`. Because it may not be obvious for people that the current example will run all combinations (which will result in a huge number of jobs if they add some python versions and OS).

---

yeah, i switched out my matrix for a "manual" that's more keeping in the style of what astropy and other affiliated packages are doing.